### PR TITLE
Fix python3.10 tests

### PR DESCRIPTION
**Pull request by mimi1vx**
https://progress.opensuse.org/issues/113513

### CODE REVIEW

**Review by grisu48 — APPROVED** [review]
LGTM! I like it that we can get rid of some old stuff that is not needed anymore.

Would way for @asmorodskyi still for approval though before merging.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[build-system]
+requires = ["setuptools >= 40.9.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "pcw"
+version = "1.0"
```
Comment (pdostal) [inline pyproject.toml:7]: It's not `1.0` but `1.13`.
Also JFYI we're using the version number for tagging containers in [ghcr.io/SUSE/pcw](https://ghcr.io/SUSE/pcw) registry.

**Review by pdostal — APPROVED** [review]
I think this is cool 🚀

File: .github/workflows/pcw.yml [hunk 2]
```diff
@@ -11,11 +11,11 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: ['3.8', '3.10']
     steps:
       - uses: actions/checkout@v2
       - name: Setup Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v3
```
Comment (asmorodskyi) [inline .github/workflows/pcw.yml:18]: does it mean that we were using python2 here or what ?

File: .github/workflows/pcw.yml [hunk 3]
```diff
@@ -11,11 +11,11 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: ['3.8', '3.10']
     steps:
       - uses: actions/checkout@v2
```
Comment (asmorodskyi) [inline .github/workflows/pcw.yml:16]: why you not changed v2 here ?

File: ocw/lib/provider.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 from datetime import datetime
 from datetime import timedelta
 from datetime import timezone
-from distutils.version import LooseVersion
+from looseversion import LooseVersion
```
Comment (asmorodskyi) [inline ocw/lib/provider.py:6]: this is super awesome ! I was expecting a lot of work with this and it appears just one-liner , cool !

File: .github/workflows/pcw.yml [hunk 5]
```diff
@@ -11,11 +11,11 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: ['3.8', '3.10']
```
Comment (asmorodskyi) [inline .github/workflows/pcw.yml:14]: I don't see any reason to keep 3.8 here , whole project already switched 3.10 . Even during testing of the code locally it is expected that people will use Docker_dev which also runs everything within python BCI container with python 3.10

File: .github/workflows/pcw.yml (hunk 2 above)
Comment (asmorodskyi) [reply .github/workflows/pcw.yml:18]: ah it is just **package** version , right ?

File: pyproject.toml (hunk 1 above)
Comment (mimi1vx) [reply pyproject.toml:7]: ahh, I got this from setup.py where was 1.0 :D

File: .github/workflows/pcw.yml (hunk 2 above)
Comment (mimi1vx) [reply .github/workflows/pcw.yml:18]: right , its action version.

File: ocw/lib/provider.py (hunk 4 above)
Comment (mimi1vx) [reply ocw/lib/provider.py:6]: unfortuqanelty new dep :(

File: ocw/lib/azure.py [hunk 6]
```diff
@@ -206,7 +206,7 @@ def cleanup_images_from_rg(self, keep_images):
                     if self.dry_run:
                         self.log_info("Deletion of image {} skipped due to dry run mode", item.name)
                     else:
-                        self.compute_mgmt_client().images.delete(self.__resource_group, item.name)
+                        self.compute_mgmt_client().images.begin_delete(self.__resource_group, item.name)
```
Comment (asmorodskyi) [inline ocw/lib/azure.py:209]: how this appears here ?  Looks like some mess in your commits tree , this change already in master  https://github.com/SUSE/pcw/commit/e6143d526c033ef19e33516e026981a41ecba5e7

File: tests/test_azure.py [hunk 7]
```diff
@@ -110,7 +110,7 @@ def mock_compute_mgmt_client(self):
         def compute_mgmt_client():
             pass
         compute_mgmt_client.images = lambda: None
-        compute_mgmt_client.images.delete = lambda rg, name: deleted_images.append(name)
+        compute_mgmt_client.images.begin_delete = lambda rg, name: deleted_images.append(name)
```
Comment (asmorodskyi) [inline tests/test_azure.py:113]: check  https://github.com/SUSE/pcw/commit/e6143d526c033ef19e33516e026981a41ecba5e7

Comment (mimi1vx) [reply tests/test_azure.py:113]: looks like something wrong with rebase ... I'll fix

**Pull request merged**